### PR TITLE
ci: Add macOS jobs

### PR DESCRIPTION
**Pull request by nirs**
Add basic tests for drenv skipping the tests requiring a test cluster
which we cannot use yet.

Github provides Intel based macOS runners supporting nested
virtualization. We can start a lima vm but it always get stuck waiting
for ssh connection to the guest. We can try to fix this later, but for
now we can run all the test that do not require a test cluster by
skipping the tests.

Since the workflow became too long split it to ramen and tools workflows:

Screenshot 2024-11-20 at 20 51 31

Screenshot 2024-11-20 at 20 51 59

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -178,17 +181,60 @@ jobs:
...
 
-  ramenctl-test:
-    name: ramenctl tests
-    runs-on: ubuntu-22.04
+  drenv-macos:
+    name: drenv
+    strategy:
+      matrix:
+        os:
+          - macos-14
```
Comment (nirs) [inline .github/workflows/ci.yaml:189]: This is a VM on M1 machine, so it does not support nested virtualization. We can try to use macos-13 (x86 vm). It it support nested virtualization we can run lima vm and run all tests.

Comment (nirs) [reply .github/workflows/ci.yaml:189]: We can use `macos-13` which is intel machine and it support nested virtualization:
https://docs.github.com/en/actions/using-github-hosted-runners/using-github-hosted-runners/about-github-hosted-runners#viewing-available-runners-for-a-repository

lima uses this to run vms on github:
https://github.com/lima-vm/lima/blob/master/.github/workflows/test.yml#L404

**Review by raghavendra-talur — CHANGES_REQUESTED** [review]
The changes themselves look good to me but the ci.yaml file has become too long now.

Please move the ramenctl and drenv tests to a different workflow file.

**Comment by nirs** [comment]
> The changes themselves look good to me but the ci.yaml file has become too long now.
> 
> Please move the ramenctl and drenv tests to a different workflow file.

Good idea, moving

**Comment by raghavendra-talur** [comment]
@nirs I will merge this once you move it out of draft.

**Pull request merged**